torch/csrc/distributed/autograd/context/context.cpp

```diff
@@ -211,6 +212,40 @@ const c10::Dict<torch::Tensor, torch::Tensor> DistAutogradContext::
   return accumulatedGrads_;
 }
 
+void DistAutogradContext::runGradCallbackForVariable(
+    const torch::autograd::Variable& variable,
+    GradCallback&& cb) {
+  std::lock_guard<std::mutex> guard(lock_);
+  auto it = accumulatedGrads_.find(variable);
+  TORCH_INTERNAL_ASSERT(
+      it != accumulatedGrads_.end(),
+      "The grad for the variable should exist in dist_autograd context.");
+  auto grad = it->value();
+  if (cb(grad)) {
+    // Needs to update the grad in the map.
+    accumulatedGrads_.insert(variable, std::move(grad));
```

Comment:
I'm doing the same as `AccumulateGrad::accumulateGrad()` does by inserting the new grad. I don't understand how it works, since the key is already in the dict and`insert()` is a no-op here. Probably, `grad` internally refers to an impl and the callback modifies the impl.

Reply:
This is a good point, I think this is a bug that is not caught by our current tests since it happens in a few edge cases: https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/functions/accumulate_grad.h#L99. I think this should be changed to `insert_or_assign`.

Reply:
@pritamdamania87 I'm going to use `insert_or_assign` in this PR. I suppose you are going to fix the existing `update_grad` callback.